identity-iota/src/credential/credential_validator.rs

```diff
@@ -0,0 +1,645 @@
+  /// - The expiration date does not meet the requirement set in `options`
+  /// - The issuance date does not meet the requirement set in `options`
+  /// - The `issuer` parameter does not provide a DID corresponding to the URL of the credential's issuer.
+  /// - The credential's signature cannot be verified using the issuer's DID Document
+  ///
+  /// Fails on the first encountered error if `fail_fast` is true, otherwise all
+  /// errors will be accumulated in the returned error.
+  // Takes &self in case this method will need some pre-computed state in the future.
+  pub fn full_validation<T: Serialize>(
+    &self,
+    credential: &Credential<T>,
+    options: &CredentialValidationOptions,
+    issuer: &ResolvedIotaDocument,
+    fail_fast: bool,
+  ) -> Result<()> {
+    self
+      .full_validation_local_error(credential, options, std::slice::from_ref(issuer), fail_fast)
+      .map_err(Error::UnsuccessfulCredentialValidation)
+  }
+
+  // This method takes a slice of issuer's instead of a single issuer in order to better accommodate presentation
+  // validation.
+  pub(super) fn full_validation_local_error<T: Serialize>(
+    &self,
+    credential: &Credential<T>,
+    options: &CredentialValidationOptions,
+    issuers: &[ResolvedIotaDocument],
+    fail_fast: bool,
+  ) -> CredentialValidationResult {
+    let signature_validation =
+      std::iter::once_with(|| Self::verify_signature_local_error(credential, issuers, &options.verifier_options));
+
+    let expiry_date_validation =
+      std::iter::once_with(|| Self::does_not_expire_before_local_error(credential, options.earliest_expiry_date));
+
+    let issuance_date_validation =
+      std::iter::once_with(|| Self::was_not_issued_after_local_error(credential, options.latest_issuance_date));
+
+    let structure_validation = std::iter::once_with(|| Self::check_structure_local_error(credential));
+
+    let validation_units_error_iter = issuance_date_validation
+      .chain(expiry_date_validation)
+      .chain(structure_validation)
+      .chain(signature_validation)
+      .filter_map(|result| result.err());
+    let validation_errors: OneOrMany<ValidationError> = if fail_fast {
+      validation_units_error_iter.take(1).collect()
+    } else {
+      validation_units_error_iter.collect()
+    };
+    if validation_errors.is_empty() {
+      Ok(())
+    } else {
+      Err(AccumulatedCredentialValidationError { validation_errors })
+    }
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use proptest::proptest;
+
+  use identity_core::common::Timestamp;
+
+  use identity_core::convert::FromJson;
+  use identity_core::crypto::KeyPair;
+  use identity_core::crypto::SignatureOptions;
+
+  use crate::credential::test_utils;
+  use crate::credential::CredentialValidationOptions;
+  use crate::document::IotaDocument;
+
+  fn deserialize_credential(credential_str: &str) -> Credential {
+    Credential::from_json(credential_str).unwrap()
+  }
```

Comment:
Nit: function seems unnecessary. It should either be inlined or replaced with some `get_credential` function which loads `SIMPLE_CREDENTIAL_JSON` rather than taking it as a parameter, since all instances use the same one.